src/cart/src/utest/test_gurt.c

```diff
@@ -1897,6 +1897,62 @@ test_gurt_atomic(void **state)
 	assert(mix  == 123456);
 }
 
+static void
+check_string_buffer(d_string_buffer_t *str_buf, int str_size, int buf_size,
+	const char *test_str)
+{
+	assert_non_null(str_buf->str);
+	assert_int_equal(str_buf->str_size ,str_size);
+	assert_int_equal(str_buf->buf_size ,buf_size);
+	if (test_str != NULL) {
+		assert_string_equal(str_buf->str, test_str);
+	}
+	d_free_string(str_buf);
+}
+
+static void
+test_gurt_string_buffer(void **state)
+{
+	int rc, i;
+	d_string_buffer_t str_buf = {0};
+
+	/* empty string */
+	rc = d_write_string_buffer(&str_buf, "");
+	assert_return_code(rc, errno);
+	check_string_buffer(&str_buf, 0, 64, NULL);
+
+	/* simple string */
+	rc = d_write_string_buffer(&str_buf, "hello there");
+	assert_return_code(rc, errno);
+	check_string_buffer(&str_buf, 11, 64, "hello there");
+
+	/* simple string append*/
+	rc = d_write_string_buffer(&str_buf, "Look");
+	assert_return_code(rc, errno);
+	rc = d_write_string_buffer(&str_buf, " ");
+	assert_return_code(rc, errno);
+ 	rc = d_write_string_buffer(&str_buf, "inside");
```

Comment:
(style)  code indent should use tabs where possible
(style)  please, no space before tabs